# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsTransportUtils.cpp
# Commit: 23b864e14db0
# Full Hash: 23b864e14db0cf874d0e126b6299c8c44922dde3
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2022-11-16 09:31:06
# Regressor Bug: 1791633
# File Overlap Count: 1
# Description:
#   Bug 1791633 - separate nsITLSSocketControl from nsITransportSecurityInfo r=necko-reviewers,kershaw,jschanck
#   
#   Depends on D160311
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160313
# ==============================================================================

diff -r 0bcba3375ec0 -r 23b864e14db0 netwerk/base/nsTransportUtils.cpp
--- a/netwerk/base/nsTransportUtils.cpp	Tue Nov 15 20:03:29 2022 +0000
+++ b/netwerk/base/nsTransportUtils.cpp	Tue Nov 15 20:03:29 2022 +0000
@@ -3,11 +3,12 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "mozilla/Mutex.h"
-#include "nsTransportUtils.h"
+#include "nsCOMPtr.h"
 #include "nsITransport.h"
 #include "nsProxyRelease.h"
+#include "nsSocketTransportService2.h"
 #include "nsThreadUtils.h"
-#include "nsCOMPtr.h"
+#include "nsTransportUtils.h"
 
 using namespace mozilla;
 
@@ -55,7 +56,17 @@
         mProgress(progress),
         mProgressMax(progressMax) {}
 
-  ~nsTransportStatusEvent() = default;
+  ~nsTransportStatusEvent() {
+    auto ReleaseTransport = [transport(std::move(mTransport))]() mutable {
+      transport = nullptr;
+    };
+    if (OnSocketThread()) {
+      ReleaseTransport();
+    } else {
+      gSocketTransportService->Dispatch(NS_NewRunnableFunction(
+          "nsHttpConnection::~nsHttpConnection", ReleaseTransport));
+    }
+  }
 
   NS_IMETHOD Run() override {
     // since this event is being handled, we need to clear the proxy's ref.